### PR TITLE
Add error when timescaledb library is not preloaded.

### PR DESCRIPTION
**Pull request by cevian**
Because this extension uses hooks, it should be preloaded into postgres.
This PR adds an error message if it is loaded dynamically (i.e. when
CREATE EXTENSION is run before without the library being preloaded).

### CODE REVIEW

**Comment by cevian** [comment]
Note the more "postgres" way would be to put the explanation in an error hint. But those are not always displayed. So in a sense this seems safer. Let me know if you disagree and think this should be in the hint field.

File: src/init.c [hunk 1]
```diff
@@ -37,6 +39,24 @@ extern void _PG_fini(void);
...
+		char	   *force_load = GetConfigOptionByName("timescaledb.force_load", NULL, true);
+		if (force_load == NULL || strlen(force_load) != 2 || strncmp(force_load, "on", 2) != 0) {
+			char	   *config_file = GetConfigOptionByName("config_file", NULL, false);
+			elog(ERROR,
+"The timescaledb library is not preloaded\n\n"
+"We strongly recommend loading the timescaledb library via shared_preload_libraries.\n\n"
+"This should be done by editing the config file at: %1$s\n"
+"and adding 'timescaledb' to the list in the shared_preload_libraries config.\n"
+"An easy way to do this, if not preloading other libraries, is with the bash command:\n"
+"	echo \"shared_preload_libraries = 'timescaledb'\" >> %1$s \n\n"
```
Comment (erimatnor) [inline src/init.c]: I guess this is fine, but might not work if this line is already uncommented, albeit set to the empty string.

File: src/init.c [hunk 2]
```diff
@@ -37,6 +39,24 @@ extern void _PG_fini(void);
 void
 _PG_init(void)
 {
+	if (!process_shared_preload_libraries_in_progress)
+	{
+	
+		char	   *force_load = GetConfigOptionByName("timescaledb.force_load", NULL, true);
+		if (force_load == NULL || strlen(force_load) != 2 || strncmp(force_load, "on", 2) != 0) {
+			char	   *config_file = GetConfigOptionByName("config_file", NULL, false);
+			elog(ERROR,
```
Comment (erimatnor) [inline src/init.c]: Consider using ereport, together with errmsg() and errhint():

```
ereport(ERROR, (errmsg("the error"),
                        errhint("hint on how to fix.")));
```

File: src/init.c [hunk 3]
```diff
@@ -37,6 +39,24 @@ extern void _PG_fini(void);
 void
 _PG_init(void)
 {
+	if (!process_shared_preload_libraries_in_progress)
+	{
+	
+		char	   *force_load = GetConfigOptionByName("timescaledb.force_load", NULL, true);
```
Comment (erimatnor) [inline src/init.c]: Confusing config option name. Sounds like this forces the extension to be loaded, while in reality it just forces a check for loaded lib. Maybe better name is `require_preload` or `enforce_preload`.

Comment (cevian) [inline src/init.c]: Actually this flag should be the opposite (if it's on then you can create extension without preload). The default should be we disallow installation without preload. maybe timescaledb.create_without_preload?

File: src/init.c (hunk 2 above)
Comment (cevian) [inline src/init.c]: Yeah I put a comment above as to whether the long text should be in the error message or the hint. The problem with hint, is it's not always displayed (according to client settings). But you think it should be in the hint?

File: src/init.c (hunk 1 above)
Comment (cevian) [inline src/init.c]: I believe last line overwrites previous entries in the config.

**Comment by olofr** [comment]
I must have missed some discussion. What is the scenario when someone would want to load the extension without the hooks installed ?

**Comment by cevian** [comment]
Mainly if they can't restart postgres and/or can't edit the config file.

File: src/init.c (hunk 3 above)
Comment (cevian) [inline src/init.c]: Changed to timescaledb.allow_install_without_preload

File: src/init.c (hunk 2 above)
Comment (cevian) [inline src/init.c]: Fixed. Changed to be a hint.

**Comment by cevian** [comment]
@olofr @erimatnor approve?

**Pull request merged**